File: Medium/0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        length = len(nums);
        if k < length:
            spot = length - k
            sub = nums[spot:]
            nums[:] = sub + nums[:spot]
        elif k > length:
            spot = length - k%length
            sub = nums[spot:]
            nums[:] = sub + nums[:spot]

File: 0189-rotate-array/0189-rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var rotate = function(nums, k) {
    if(nums.length <= k){
        k = k % nums.length
    }
    let tmp = nums.splice(-k)

    nums.splice(0,0,...tmp)
};